boards/arm/mps4/doc/index.rst

```diff
@@ -0,0 +1,188 @@
+   for building for both Secure and Non-Secure firmware.
+
+   The BOARD options are summarized below:
+
+   +-------------------------------+-----------------------------------------------+
+   |   BOARD                       | Description                                   |
+   +===============================+===============================================+
+   | ``mps4/corstone320/fvp``      | For building Secure (or Secure-only) firmware |
+   +-------------------------------+-----------------------------------------------+
+   | ``mps4/corstone320/fvp/ns``   | For building Non-Secure firmware              |
+   +-------------------------------+-----------------------------------------------+
+
+   FPGA Usage:
+    - N/A.
+
+   FVP Usage:
+    - To run with the FVP, first set environment variable ``ARMFVP_BIN_PATH`` before using it. Then you can run it with ``west build -t run``.
+
+    .. code-block:: bash
+
+       export ARMFVP_BIN_PATH=/path/to/fvp/directory
+       west build -b {BOARD qualifier from table above} samples/hello_world -t run
+
+   To run the Fixed Virtual Platform simulation tool you must download "FVP model
+   for the Corstone-320 MPS4" from Arm and install it on your host PC. This board
+   has been tested with version 11.27.25 (Sep 24 2024).
+
+   QEMU Usage:
+    - N/A.
+
+.. note::
+
+   - Board qualifier must include the variant name as mentioned above.
+     ``mps4/corstone320`` without the variant name is not a valid qualifier.
+   - ``mps4/corstone320/fvp/ns`` variant needs latest upstream TF-M release since Zephyr's current
+     TF-M doesn't support Corstone-320 FVP yet.
+
+Hardware
+********
+
+ARM MPS4 provides the following hardware components:
+
+- CPU
+
+  - Support Arm Cortex-M85 CPU and
+    Soft Macro Model (SMM) implementation of SSE-320 subsystem
+
+- Memory
+
+  - 8GB DDR4 SODIMM, with support for up to 16 GB single-rank or dual-rank RAM
+  - A user microSD card that supports 1-bit SPI or 4-bit QSPI controllers
+  - 128MB QSPI Flash
+  - The Ultrascale+ VU13P FPGA has 11.8MB of FPGA Block RAM (BRAM) and up to 45 MB UltraRam (URAM)
+
+- Debug
+
+  - The MPS4 board provides connectors that support P-JTAG processor debug, F-JTAG FPGA debug, 16-bit and 4-bit trace, and Serial Wire Debug (SWD):
+
+    - 60-pin MIPI debug connector (J18)
+    - 20-pin CoreSight™ debug and ETM connector (J19)
+    - 10-pin CoreSight™ debug connector (J20)
+    - 14-pin FPGA ILA debug connector (J22)
+
+- Interface
+
+  - AHB GPIO
+  - UART
+  - SPI
+  - I2C
+  - I2S
+  - Color LCD serial interface
+  - Ethernet
+
+- On-board Peripherals
+
+  - Color LCD
+  - 10 LEDs
+  - 8 Switches
+  - 2 user push buttons
+
+Supported Features
+===================
+
+.. zephyr:board-supported-hw::
+
+.. note::
+
+   The default configuration can be found in :zephyr_file:`boards/arm/mps4/mps4_corstone320_fvp_defconfig`.
+   See the `MPS4 FPGA Website`_ for a complete list of MPS4 board hardware features.
```

Comment:
no need to link to defconfig file, the board doc page already has a "Browse board sources" link

```suggestion
See the `MPS4 FPGA Website`_ for a complete list of MPS4 board hardware features.
```